### PR TITLE
Added comments, cleaned up the code a bit and updated README

### PR DESCRIPTION
**Pull request by paulpela**

### CODE REVIEW

**Comment by marekpridal** [comment]
Thanks a lot for contribution! I'll take a look as soon as possible 🙂

File: Sources/BarChartKit/SwiftUI/BarChart.swift [hunk 1]
```diff
@@ -1,40 +1,38 @@
 import SwiftUI
 
+/// Creates a Bar Chart similar to the chart used in iOS Health app
 public struct BarChart: View {
-
     // MARK: Structs
+    /// Provides data for the BarChart
     public struct DataSet {
+        /// Single data element for BarChart. Can consist of multiple bars
         public struct DataElement: Equatable, Identifiable {
```
Comment (marekpridal) [inline Sources/BarChartKit/SwiftUI/BarChart.swift:9]: Move please conformance of protocols to extensions where implementation is.

File: Sources/BarChartKit/SwiftUI/BarChart.swift [hunk 2]
```diff
@@ -104,6 +110,30 @@ public struct BarChart: View {
     }
 }
 
+extension BarChart.DataSet.DataElement {
```
Comment (marekpridal) [inline Sources/BarChartKit/SwiftUI/BarChart.swift:113]: Make extensions public, too.

**Review by marekpridal — CHANGES_REQUESTED** [review]
Thanks a lot for improving documentation and code cleanest! I added few comments. Also I noticed GitHub Actions seems to be disabled in you profile and because of that the CI check cannot pass. Can you please enable it to be sure, everything works as expected?

**Comment by paulpela** [comment]
Hi, I fixed conformance locations and added public wherever needed, however I can't fix the checks issue. Actions are definitely enabled on my repository. Any ideas?

**Comment by paulpela** [comment]
Looks like it worked now! Sorry for the mess! I learned a lot here :)

**Review by marekpridal — APPROVED** [review]
Thanks a lot for update!

**Pull request merged**